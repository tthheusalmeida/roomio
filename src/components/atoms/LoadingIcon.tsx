import { VscLoading } from "react-icons/vsc";

export default function LoadingIcon() {
  return <VscLoading className="animate-spin" size={24} />;
}
