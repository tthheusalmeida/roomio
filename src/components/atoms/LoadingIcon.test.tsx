import { render, screen } from "@testing-library/react";
import LoadingIcon from "./LoadingIcon";

jest.mock("react-icons/vsc", () => ({
  VscLoading: (props: any) => <span data-testid="loading-icon" {...props} />,
}));

describe("LoadingIcon component", () => {
  it("renders the loading icon with correct class", () => {
    const { asFragment } = render(<LoadingIcon />);

    const icon = screen.getByTestId("loading-icon");
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass("animate-spin");

    expect(asFragment()).toMatchSnapshot();
  });
});
