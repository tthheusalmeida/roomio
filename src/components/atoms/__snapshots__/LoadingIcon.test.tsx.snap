// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`LoadingIcon component renders the loading icon with correct class 1`] = `
<DocumentFragment>
  <span
    class="animate-spin"
    data-testid="loading-icon"
    size="24"
  />
</DocumentFragment>
`;
